conda_self/cli/main_protect.py

```diff
@@ -10,9 +11,76 @@
 
 def configure_parser(parser: argparse.ArgumentParser) -> None:
     parser.description = HELP
+    parser.add_argument(
+        "--default-env",
+        action="store",
+        default="default",
+        help="Name of the new default environment",
+    )
     parser.set_defaults(func=execute)
 
 
 def execute(args: argparse.Namespace) -> int:
+    from contextlib import redirect_stdout
+
+    from conda.base.context import context, sys_rc_path
+    from conda.cli.main_config import _read_rc, _write_rc
+    from conda.cli.main_list import print_explicit
+    from conda.core.prefix_data import PrefixData
+    from conda.misc import clone_env
+    from conda.reporters import confirm_yn
+
+    from ..query import permanent_dependencies
+    from ..reset import reset
+
     print("Protecting 'base' environment...")
+    uninstallable_packages = permanent_dependencies()
+
+    # Ensure the destination default environment does not exist already
+    dest_prefix_data = PrefixData.from_name(args.default_env)
+    if dest_prefix_data.is_environment():
+        confirm_yn(
+            "WARNING: A conda environment already exists at "
+            "'{dest_prefix_data.prefix_path}'\n\n"
+            "Remove existing environment?\nThis will remove ALL packages "
+            "contained within this specified prefix directory.\n\n",
+            default="no",
+            dry_run=False,
+        )
+        reset(prefix=dest_prefix_data.prefix_path)
```

Comment:
This will empty out the environment but might leave files behind (e.g. `pip install` etc), have we checked it works well in that case?

Reply:
I haven't checked if reset is able to deal with pip packages. I expect it's not going to work though. I also expect cloning to not work quite right with pip packages. 

I think a quick solution might be to do a reset and delete all the files in the prefix.

But, if working with pip is really important for this functionality, it might be good to dig into the problem more. What do you think?

Reply:
I think it'll be important in the future, but we don't officially support pip installs in `conda` so far, so we don't have to do it now.

> I think a quick solution might be to do a reset and delete all the files in the prefix.

Let's do this!